polynote-kernel/src/main/scala/polynote/kernel/remote/DeploySparkSubmit.scala

```diff
@@ -41,8 +41,15 @@ object DeploySparkSubmit extends DeployCommand {
     val sparkArgs = (sparkConfig - "sparkSubmitArgs" - "spark.driver.extraJavaOptions" - "spark.submit.deployMode" - "spark.driver.memory")
       .flatMap(kv => Seq("--conf", s"${kv._1}=${kv._2}"))
 
+    val versionConfig = nbConfig.scalaVersion.flatMap(scalaVersion => {
+      nbConfig.sparkTemplate
+        .map(_.versionConfigs)
+        .flatMap(versionConfig => versionConfig.flatMap(_.find(_.versionNumber == scalaVersion)))
+    })
+
     val sparkSubmitArgs =
       nbConfig.sparkTemplate.flatMap(_.sparkSubmitArgs).toList.flatMap(parseQuotedArgs) ++
+      versionConfig.map(_.sparkSubmitArgs).toList.flatMap(parseQuotedArgs) ++
```

Comment:
Hmm, so what happens if someone specifies `spark_submit_args` in the old place and also inside their `versions_configs`? IIUC the values will be concatenated, right? 

I think it might be better to fail in that case: if you have `version_configs` then the args can only be set inside. WDYT?

Reply:
Good point, concatenation probably isn't great. Erroring out seems fine to me, another option could be having `version_configs` just override `spark_submit_args` for that template? Seems unlikely to me in either case that a user would have both set.